_stbt/mask.py

```diff
@@ -79,24 +86,45 @@ def __init__(self, m, *, invert=False):
             raise TypeError("Expected filename, Image, Mask, or Region. "
                             f"Got {m!r}")
 
-        # for memoisation:
-        self._array = None
+    def __eq__(self, o):
+        if not isinstance(o, Mask):
+            return False
+        if self._array is not None:
+            return numpy.array_equal(self._array, o._array)
+        else:
+            return ((self._filename, self._binop, self._region, self._invert) ==
+                    (o._filename, o._binop, o._region, o._invert))
+
+    def __hash__(self):
+        if self._array is not None:
+            from _stbt.xxhash import Xxhash64
+            h = Xxhash64()
+            h.update(numpy.ascontiguousarray(self._array).data)
+            return hash((self._array.shape, h.hexdigest(), self._invert))
+        else:
+            return hash((self._filename, self._binop, self._region,
+                         self._invert))
```

Comment:
With this implementation two `Mask`s can be identical, but constructed differently and so hash differently.  Maybe that's fine - as long as we preserve consistency between `__eq__` and `__hash__`.  The alternative would be to have a wrapper type to pass through `lru_cache` that provides the properties that we require in this instance, but not public.